### PR TITLE
Rep 2769 packaging

### PR DESCRIPTION
**Pull request by dkowis**
Build a deb package, including an init script and a default config file.

The init script could use a bit of manual testing (or maybe a vagrant based test in this project?) but I ran out of time to get this into you guys.

### CODE REVIEW

**Comment by dmnjohns** [comment]
Looks good, but I need to test it. Where did all the time go?!

Isn't the os-package plugin great? =)

**Comment by dkowis** [comment]
Where did the time go?

File: src/files/inits/repose-phone-home-debian.init [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    update-rc.d -f repose-phone-home remove
+    rm -fv "$0"
+  fi
+}
+
+status() {
+        printf "%-50s" "Checking $NAME..."
+    if [ -f $PIDFILE ]; then
+        PID=$(cat $PIDFILE)
+            if [ -z "$(ps axf | grep ${PID} | grep -v grep)" ]; then
```
Comment (dmnjohns) [inline src/files/inits/repose-phone-home-debian.init]:
```
vagrant@precise32:~$ sudo service repose-phone-home status
Checking repose-phone-home...                     Usage: grep [OPTION]... PATTERN [FILE]...
Try `grep --help' for more information.
The process appears to be dead but pidfile still exists
```

So I get that message in my Ubuntu VM.

```
vagrant@precise32:~$ uname -a
Linux precise32 3.2.0-23-generic-pae #36-Ubuntu SMP Tue Apr 10 22:19:09 UTC 2012 i686 i686 i386 GNU/Linux
```

Comment (dkowis) [inline src/files/inits/repose-phone-home-debian.init]: I did copypasta the heck out of this. I'll work on it tonight, get it cleaned up.

File: src/files/inits/repose-phone-home-debian.init [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+start() {
+  if [ -f $PIDFILE ] && kill -0 $(cat $PIDFILE); then
+    echo 'Service already running' >&2
+    return 1
+  fi
+  echo 'Starting service…' >&2
+  local CMD="$SCRIPT &> \"$LOGFILE\" & echo \$!"
+  su -c "$CMD" $RUNAS > "$PIDFILE"
+  echo 'Service started' >&2
+}
```
Comment (dmnjohns) [inline src/files/inits/repose-phone-home-debian.init]:
```
vagrant@precise32:~$ sudo service repose-phone-home start
/etc/init.d/repose-phone-home: 19: kill: Usage: kill [-s sigspec | -signum | -sigspec] [pid | job]... or
kill -l [exitstatus]
Starting service…
Unknown id: phone-home
Service started
vagrant@precise32:~$ sudo service repose-phone-home status
Checking repose-phone-home...                     Usage: grep [OPTION]... PATTERN [FILE]...
Try `grep --help' for more information.
The process appears to be dead but pidfile still exists
```

That's what I get when I try to start the service. Log file does not appear to get created. Service does not start.

**Comment by dmnjohns** [comment]
The good news is, all of the files appear to go where they are supposed to go!

The bad news -- permissions seem borked:
```
vagrant@precise32:~$ dpkg -c /vagrant/repose-phone-home_1.0-1_i386.deb 
drwxr-xr-x dami6356/0        0 2015-10-16 15:47 ./opt/
drwxr-xr-x dami6356/0        0 2015-10-16 15:47 ./opt/repose-phone-home/
drwxr-xr-x dami6356/0        0 2015-10-16 15:47 ./opt/repose-phone-home/bin/
-rwxr-xr-x dami6356/0     6860 2015-10-16 15:47 ./opt/repose-phone-home/bin/repose-phone-home-service
-rwxr-xr-x dami6356/0     4267 2015-10-16 15:47 ./opt/repose-phone-home/bin/repose-phone-home-service.bat
drwxr-xr-x dami6356/0        0 2015-10-16 15:47 ./opt/repose-phone-home/lib/
-rw-r--r-- dami6356/0     4467 2015-10-16 15:47 ./opt/repose-phone-home/lib/aopalliance-1.0.jar
-rw-r--r-- dami6356/0    62873 2015-10-16 15:47 ./opt/repose-phone-home/lib/classmate-1.1.0.jar
-rw-r--r-- dami6356/0   692267 2015-10-16 15:47 ./opt/repose-phone-home/lib/hibernate-validator-5.2.1.Final.jar
-rw-r--r-- dami6356/0    46968 2015-10-16 15:47 ./opt/repose-phone-home/lib/jackson-annotations-2.6.0.jar
-rw-r--r-- dami6356/0   258833 2015-10-16 15:47 ./opt/repose-phone-home/lib/jackson-core-2.6.1.jar
-rw-r--r-- dami6356/0  1166488 2015-10-16 15:47 ./opt/repose-phone-home/lib/jackson-databind-2.6.1.jar
-rw-r--r-- dami6356/0    66711 2015-10-16 15:47 ./opt/repose-phone-home/lib/jboss-logging-3.2.1.Final.jar
-rw-r--r-- dami6356/0    16618 2015-10-16 15:47 ./opt/repose-phone-home/lib/jcl-over-slf4j-1.7.12.jar
-rw-r--r-- dami6356/0     4736 2015-10-16 15:47 ./opt/repose-phone-home/lib/jul-to-slf4j-1.7.12.jar
-rw-r--r-- dami6356/0    24567 2015-10-16 15:47 ./opt/repose-phone-home/lib/log4j-over-slf4j-1.7.12.jar
-rw-r--r-- dami6356/0   280928 2015-10-16 15:47 ./opt/repose-phone-home/lib/logback-classic-1.1.3.jar
-rw-r--r-- dami6356/0   455041 2015-10-16 15:47 ./opt/repose-phone-home/lib/logback-core-1.1.3.jar
-rw-r--r-- dami6356/0   604996 2015-10-16 15:47 ./opt/repose-phone-home/lib/mongo-java-driver-2.13.2.jar
-rw-r--r-- dami6356/0     5407 2015-10-16 15:47 ./opt/repose-phone-home/lib/repose-phone-home-service-1.0.jar
-rw-r--r-- dami6356/0  5745606 2015-10-16 15:47 ./opt/repose-phone-home/lib/scala-library-2.11.7.jar
-rw-r--r-- dami6356/0    32127 2015-10-16 15:47 ./opt/repose-phone-home/lib/slf4j-api-1.7.12.jar
-rw-r--r-- dami6356/0   269579 2015-10-16 15:47 ./opt/repose-phone-home/lib/snakeyaml-1.16.jar
-rw-r--r-- dami6356/0   366503 2015-10-16 15:47 ./opt/repose-phone-home/lib/spring-aop-4.2.1.RELEASE.jar
-rw-r--r-- dami6356/0   729634 2015-10-16 15:47 ./opt/repose-phone-home/lib/spring-beans-4.2.1.RELEASE.jar
-rw-r--r-- dami6356/0   504593 2015-10-16 15:47 ./opt/repose-phone-home/lib/spring-boot-1.3.0.M5.jar
-rw-r--r-- dami6356/0   691939 2015-10-16 15:47 ./opt/repose-phone-home/lib/spring-boot-autoconfigure-1.3.0.M5.jar
-rw-r--r-- dami6356/0     2200 2015-10-16 15:47 ./opt/repose-phone-home/lib/spring-boot-starter-1.3.0.M5.jar
-rw-r--r-- dami6356/0     2240 2015-10-16 15:47 ./opt/repose-phone-home/lib/spring-boot-starter-data-mongodb-1.3.0.M5.jar
-rw-r--r-- dami6356/0     2234 2015-10-16 15:47 ./opt/repose-phone-home/lib/spring-boot-starter-logging-1.3.0.M5.jar
-rw-r--r-- dami6356/0     2199 2015-10-16 15:47 ./opt/repose-phone-home/lib/spring-boot-starter-tomcat-1.3.0.M5.jar
-rw-r--r-- dami6356/0     2225 2015-10-16 15:47 ./opt/repose-phone-home/lib/spring-boot-starter-validation-1.3.0.M5.jar
-rw-r--r-- dami6356/0     2208 2015-10-16 15:47 ./opt/repose-phone-home/lib/spring-boot-starter-web-1.3.0.M5.jar
-rw-r--r-- dami6356/0  1084804 2015-10-16 15:47 ./opt/repose-phone-home/lib/spring-context-4.2.1.RELEASE.jar
-rw-r--r-- dami6356/0  1067974 2015-10-16 15:47 ./opt/repose-phone-home/lib/spring-core-4.2.1.RELEASE.jar
-rw-r--r-- dami6356/0   635520 2015-10-16 15:47 ./opt/repose-phone-home/lib/spring-data-commons-1.11.0.RELEASE.jar
-rw-r--r-- dami6356/0   654204 2015-10-16 15:47 ./opt/repose-phone-home/lib/spring-data-mongodb-1.8.0.RELEASE.jar
-rw-r--r-- dami6356/0   262382 2015-10-16 15:47 ./opt/repose-phone-home/lib/spring-expression-4.2.1.RELEASE.jar
-rw-r--r-- dami6356/0   252964 2015-10-16 15:47 ./opt/repose-phone-home/lib/spring-tx-4.1.7.RELEASE.jar
-rw-r--r-- dami6356/0   768517 2015-10-16 15:47 ./opt/repose-phone-home/lib/spring-web-4.2.1.RELEASE.jar
-rw-r--r-- dami6356/0   866968 2015-10-16 15:47 ./opt/repose-phone-home/lib/spring-webmvc-4.2.1.RELEASE.jar
-rw-r--r-- dami6356/0  2812733 2015-10-16 15:47 ./opt/repose-phone-home/lib/tomcat-embed-core-8.0.26.jar
-rw-r--r-- dami6356/0   238027 2015-10-16 15:47 ./opt/repose-phone-home/lib/tomcat-embed-el-8.0.26.jar
-rw-r--r-- dami6356/0    40824 2015-10-16 15:47 ./opt/repose-phone-home/lib/tomcat-embed-logging-juli-8.0.26.jar
-rw-r--r-- dami6356/0   243648 2015-10-16 15:47 ./opt/repose-phone-home/lib/tomcat-embed-websocket-8.0.26.jar
-rw-r--r-- dami6356/0    63777 2015-10-16 15:47 ./opt/repose-phone-home/lib/validation-api-1.1.0.Final.jar
drwxr-xr-x dami6356/0        0 2015-10-16 15:47 ./etc/
drwxr-xr-x dami6356/0        0 2015-10-16 15:47 ./etc/repose-phone-home/
-rw-r--r-- dami6356/0      122 2015-10-16 15:47 ./etc/repose-phone-home/sample-config.cfg
-r-xr-x--- dami6356/0     1976 2015-10-16 15:47 ./opt/repose-phone-home/bin/repose-phone-home-debian.init
drwxr-xr-x root/root         0 2015-10-16 15:47 ./etc/init.d/
lrw-r--r-- root/root         0 2015-10-16 15:47 ./etc/init.d/repose-phone-home -> /opt/repose-phone-home/bin/repose-phone-home-debian.init
```

I ran the Gradle build outside of the VM, so I'm guessing since ownership wasn't defined in the os-package part of the build, the default owner is the user who built the package. Probably not what we want.

**Comment by dkowis** [comment]
It should install owned by root now, set up the proper user, and the init script starts the app :raised_hands:

**Pull request merged**